src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -57,11 +57,7 @@
 import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
 import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
 import org.glassfish.hk2.api.MultiException;
```

Comment:
The only place the MultiException is used is in an multi-catch with ExecutionException which is what gets thrown by either implementation from what I can tell in the `.get()` call because it's async. In my tests it seems to propagate exceptions just fine. They get dealt with by the line `} else if (cause instanceof WebApplicationException) {...`